core/src/main/java/com/expediagroup/streamplatform/streamregistry/core/services/ConsumerBindingService.java

```diff
@@ -113,4 +113,17 @@ public boolean exists(ConsumerBindingKey key) {
     ), null, null);
     return consumerBindingRepository.findAll(example).stream().findFirst();
   }
+
+  @PostAuthorize("returnObject.isPresent() ? hasPermission(returnObject, 'READ') : true")
```

Comment:
Question, it's been a while since I used the Spring Security annotations, but for a void function, I'd have thought that `returnObject` is always null and so this will always return `true`.

Not sure if there is a global permission we can use here in conjuction with `PreAuthorize` instead?

Reply:
I'm still not sure this makes sense being a `PostAuthorize`. After the function has run, we've already run the delete so even authorization failed the delete would have been processed.

In my mind a `PostAuthorize` check would make more sense here

Reply:
shouldn't this check for a DELETE permission instead of READ?

Reply:
updated to use preauthorize and 'delete'